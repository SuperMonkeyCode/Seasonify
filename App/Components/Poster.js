import React from 'react';
import FitImage from 'react-native-fit-image';

const Poster = ({ url }) => (
  <FitImage
    source={{ uri: url }}
    resizeMode="contain"
    resizeMethod="scale"
    indicator
    indicatorColor="#919191"
    indicatorSize="large"
  />
);

export default Poster;
