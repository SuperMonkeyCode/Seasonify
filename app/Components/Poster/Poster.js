/* @flow */
import { Image } from 'react-native';
import { compose, pure } from 'recompose';
import React from 'react';

import withResponsive from 'HOC/withResponsive';
import Metrics from 'Themes/Metrics';

type Props = {
  /* parent */
  url: string,
  orientation: 'PORTRAIT' | 'LANDSCAPE',
  window: Object,
  /* connect */
  /* HOC */
};

const enhance = compose(pure, withResponsive);

function Poster({ url, orientation, window }: Props) {
  const imageWidth = window.width * Metrics.columnLeft / 100;

  return (
    <Image
      source={{ uri: url }}
      resizeMode="cover"
      resizeMethod="scale"
      orientation={orientation}
      style={{
        height: imageWidth * Metrics.poster.averageRatio,
        width: imageWidth,
      }}
    />
  );
}

export default enhance(Poster);
