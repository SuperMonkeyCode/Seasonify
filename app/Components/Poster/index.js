/* @flow */
export { default } from './Poster';
