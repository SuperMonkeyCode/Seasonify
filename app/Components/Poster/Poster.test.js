import { Dimensions, Image } from 'react-native';
import React from 'react';

import Poster from './Poster';

function setup(specificProps = {}) {
  const props = {
    url: 'http://somewhere.net/poster.jpg',
    ...specificProps,
  };
  const component = shallowDive(<Poster {...props} />, Image);
  return {
    component,
    props,
  };
}

describe('Rendering on default screen', () => {
  it('should render as PORTRAIT', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});

describe('Rendering on landscape screen', () => {
  let component;

  beforeEach(() => {
    Dimensions.get = jest.fn().mockReturnValue({
      fontscale: 2,
      scale: 2,
      width: 1260,
      heigh: 720,
    });
    Dimensions.addEventListener = jest.fn();
    const { component: c } = setup();
    component = c;
  });

  it('should render as LANDSCAPE', () => {
    expect(component.props().orientation).toEqual('LANDSCAPE');
    expect(component).toMatchSnapshot();
  });
});
