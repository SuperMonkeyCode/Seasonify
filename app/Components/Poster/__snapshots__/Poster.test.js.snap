// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering on default screen should render as PORTRAIT 1`] = `
<Image
  orientation="PORTRAIT"
  resizeMethod="scale"
  resizeMode="cover"
  source={
    Object {
      "uri": "http://somewhere.net/poster.jpg",
    }
  }
  style={
    Object {
      "height": 380,
      "width": 285,
    }
  }
/>
`;

exports[`Rendering on landscape screen should render as LANDSCAPE 1`] = `
<Image
  orientation="LANDSCAPE"
  resizeMethod="scale"
  resizeMode="cover"
  source={
    Object {
      "uri": "http://somewhere.net/poster.jpg",
    }
  }
  style={
    Object {
      "height": 638.4,
      "width": 478.8,
    }
  }
/>
`;
